software/image_processor.py

```diff
@@ -136,20 +183,63 @@ def get_lines(self, image):
         # calculates the line equations
         for line in lines:
             x1, y1, x2, y2 = line[0]
-            cv2.line(copyimg, (x1, y1), (x2, y2), (255, 0, 0), 1)
+            midx = int((x1 + x2) / 2)
+            midy = int((y1+y2)/2)
+            robot_out = False
             
+
             if x1 == x2:
                 continue
             slope = (y2 - y1) / (x2 - x1) # slope
 
-            if (slope < -3 or slope > 3):
+            #antislope = 1/slope
+
+            # LINE COLOUR SAMPLING
+            w_counter = 0
+            b_counter = 0
+            for i in range(10):
+                if (midy + i) >= 480 or (midy + i) < 0:
+                    continue
+                if (midx >= 640 or midx < 0):
+                    continue
+                if fragmentedwhite[midy + i, midx] != 0:
+                    w_counter += 1
+                if fragmentedblack[midy - i, midx] != 0:
+                    b_counter += 1
+            if b_counter < 2 or w_counter < 2:
+                b_counter = 0
+                w_counter = 0
+                # CHECK IF ROBOT IS OUTSIDE OF LINE
+#                for j in range(10):
```

Comment:
Are you planning to use this code? Why to keep it as commented-out? Git provides version management and 'looking into past'

Reply:
Fixed